interpreter/interpreter_test.go

```diff
@@ -80,17 +80,81 @@ func TestInterpreter_ConditionalExpr(t *testing.T) {
 }
 
 func TestInterpreter_ComprehensionExpr(t *testing.T) {
-	// [1, 1u, 1.0].exists(x, type(x) == uint)
-	program := NewProgram(
-		test.Exists.Expr,
-		test.Exists.Info(t.Name()))
-
-	interpretable := interpreter.NewInterpretable(program)
+	parsed, errors := parser.ParseText("[1, 1u, 1.0].exists(x, type(x) == uint)")
+	if len(errors.GetErrors()) != 0 {
+		t.Errorf(errors.ToDisplayString())
+	}
+	pgrm := NewProgram(parsed.Expr, parsed.SourceInfo)
+	eval := interpreter.NewInterpretable(pgrm)
 	// TODO: make the type identifiers part of the standard declaration set.
-	result, _ := interpretable.Eval(
-		NewActivation(map[string]interface{}{}))
+	result, _ := eval.Eval(emptyActivation)
 	if result != types.True {
-		t.Errorf("Expected true, got: %v", result)
+		t.Errorf("Got %v, wanted true", result)
+	}
+}
+
+func TestInterpreter_NonStrictExistsComprehension(t *testing.T) {
+	parsed, errors := parser.ParseText("[0, 2, 4].exists(x, 4/x == 2 && 4/(4-x) == 2)")
+	if len(errors.GetErrors()) != 0 {
+		t.Errorf(errors.ToDisplayString())
+	}
+	pgrm := NewProgram(parsed.Expr, parsed.SourceInfo)
+	eval := interpreter.NewInterpretable(pgrm)
+	result, _ := eval.Eval(emptyActivation)
+	if result != types.True {
+		t.Errorf("Got %v, wanted true", result)
+	}
+}
```

Comment:
You're using this pattern enough that it seems to be worth abstracting:

func TestInterpreter_Blah(t *testing.T) {
        EvalExpectingTrue("[0, 2, 4].exists(x, whatever(x))")
}